tests/Doctrine/Tests/DBAL/Query/Expression/CompositeExpressionTest.php

```diff
@@ -77,4 +77,12 @@ public function provideDataForConvertToString()
             ),
         );
     }
+
+    public function testAddPreventsAddingEmptyCompositeExpressionObjects()
+    {
+        $emptyExpr = new CompositeExpression(CompositeExpression::TYPE_AND, []);
+        $expr = new CompositeExpression(CompositeExpression::TYPE_AND, [$emptyExpr]);
+
+        $this->assertEquals(0, count($expr));
```

Comment:
Hehe sorry if I was not clear enough. What I want to point out is that if passing an empty instance of `CompositeExpression` it should not be rejected at all IMO because it is passed as an object by reference and you can still add parts to that later on. Example:

```php
$expression1 = new CompositeExpression(CompositeExpression::TYPE_AND, []);
$expression2 = new CompositeExpression(CompositeExpression::TYPE_AND, [$expression1]);

$expression1->add(array('u.group_id = 1')); // now $expression1 is not empty anymore
```

This is what I was referring to as "the API is mutable".